streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -280,193 +270,199 @@
                     Type.LIST,
                     Importance.HIGH,
                     CommonClientConfigs.BOOTSTRAP_SERVERS_DOC)
-            .define(CLIENT_ID_CONFIG,
+            .define(REPLICATION_FACTOR_CONFIG,
+                    Type.INT,
+                    1,
+                    Importance.HIGH,
+                    REPLICATION_FACTOR_DOC)
+            .define(STATE_DIR_CONFIG,
                     Type.STRING,
-                    "",
+                    "/tmp/kafka-streams",
                     Importance.HIGH,
-                    CommonClientConfigs.CLIENT_ID_DOC)
+                    STATE_DIR_DOC)
             .define(ZOOKEEPER_CONNECT_CONFIG,
```

Comment:
I'm wondering if we should move this config to LOW as it is already deprecated and hence would suggest users to never change it any more.